crates/core_types/src/dna/mod.rs

```diff
@@ -257,6 +257,38 @@ impl Dna {
             .flatten()
             .collect()
     }
+
+    // Check that all the zomes in the DNA have code with the required callbacks
+    // TODO: Add more advanced checks that actually try and call required functions
+    pub fn verify(&self) -> HcResult<()> {
+        let errors: Vec<HolochainError> = self
+            .zomes
+            .iter()
+            .map(|(zome_name, zome)| {
+                // currently just check the zome has some code
+                if zome.code.code.len() > 0 {
+                    Ok(())
+                } else {
+                    Err(HolochainError::ErrorGeneric(format!(
+                        "Zome {} has no code!",
+                        zome_name
+                    )))
+                }
+            })
+            .filter_map(|r| match r {
+                Ok(_) => None,
+                Err(e) => Some(e.to_owned()),
```

Comment:
r.err() will return an option so we could probably change this to filter_map(|r| r.err())

Reply:
Nice!